src/linux/backend-v4l2.cpp

```diff
@@ -1181,53 +1182,82 @@ namespace librealsense
                                 }
                                 else
                                 {
-                                    if (!is_metadata_streamed()) // when metadata is not enabled at all, streaming only video
+                                    if (!_info.has_metadata_node)
                                     {
-                                        auto timestamp = (double)buf.timestamp.tv_sec * 1000.f + (double)buf.timestamp.tv_usec / 1000.f;
-                                        timestamp = monotonic_to_realtime(timestamp);
-
-                                        LOG_DEBUG_V4L("no metadata streamed");
-                                        if (buf_mgr.verify_vd_md_sync())
+                                        if(has_metadata())
                                         {
-                                            buffer->attach_buffer(buf);
-                                            buf_mgr.handle_buffer(e_video_buf, -1); // transfer new buffer request to the frame callback
+                                            auto timestamp = (double)buf.timestamp.tv_sec*1000.f + (double)buf.timestamp.tv_usec/1000.f;
+                                            timestamp = monotonic_to_realtime(timestamp);
+
+                                            // Read metadata. Metadata node performs a blocking call to ensure video and metadata sync
+                                            acquire_metadata(buf_mgr,fds,compressed_format);
+                                            md_extracted = true;
 
+                                            if (wa_applied)
```

Comment:
Can you add a comment about this wa?
Why do we need it? why do we add 28 bytes

Reply:
This is legacy code - only moved up or down.